src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -283,7 +297,14 @@ public double getLoadTargetQ() {
 
     @Override
     public void setLoadTargetQ(double loadTargetQ) {
-        this.loadTargetQ = loadTargetQ * PerUnit.SB;
+        double newLoadTargetQ = loadTargetQ * PerUnit.SB;
+        if (newLoadTargetQ != this.loadTargetQ) {
```

Comment:
I realized this old/new test is only there because of restoring all bus states instead of only those which have changed, maybe I should have a look there if there's something to do. and if it's worth it.

Reply:
Indeed we could probably check if restoration is relevant on BusState/BranchState side but in any case I think it is safer to add this guard here.